pkg/protostore/protostore_test.go

```diff
@@ -304,20 +303,38 @@ func testProtoStoreFindAll(t *testing.T, ctx context.Context) {
 	{
 		// Check if you can find all memory records when passing a pointer to slice of pointers to structs
 		var actualQueryResults MemoryPtrSlice = make([]*pb.Memory, 0)
-		metadataMap, err := p.FindAll(ctx, &actualQueryResults)
+		metadataMap, err := p.FindAll(ctx, &actualQueryResults, datastore.NoPagination())
 		assert.NoError(err)
 		assert.Len(actualQueryResults, 2)
 		assert.Len(metadataMap, 2)
 
 		sort.Sort(actualQueryResults)
 		assert.Equal(expectedMemoryQueryResults[0].String(), actualQueryResults[0].String())
 		assert.Equal(expectedMemoryQueryResults[1].String(), actualQueryResults[1].String())
+
+		actualQueryResults = make([]*pb.Memory, 0)
+		page1 := make([]*pb.Memory, 0)
+		page2 := make([]*pb.Memory, 0)
+		metadataMap, err = p.FindAll(ctx, &page1, datastore.GetPagination(0, 1, ""))
+		assert.NoError(err)
+		assert.Len(page1, 1)
+		assert.Len(metadataMap, 1)
+		actualQueryResults = append(actualQueryResults, page1[0])
+		metadataMap, err = p.FindAll(ctx, &page2, datastore.GetPagination(1, 1, ""))
+		assert.NoError(err)
+		assert.Len(page2, 1)
+		assert.Len(metadataMap, 1)
+		actualQueryResults = append(actualQueryResults, page2[0])
+
+		sort.Sort(actualQueryResults)
+		assert.Equal(expectedMemoryQueryResults[0].String(), actualQueryResults[0].String())
+		assert.Equal(expectedMemoryQueryResults[1].String(), actualQueryResults[1].String())
```

Comment:
I propose moving these lines to a separate block, since this block initially was meant to demonstrate that finding all memory records when passing a pointer to slice of pointers to structs works. The new block could have comments that would say that they are demonstating how pagination works.

And for demonstrating pagination support, I think it would be better if there was a _for_ loop, which would run until `FindAll` can't find anything more (returns an empty slice). At each iteration, it would increment pagination offset by pagination limit (since in a real scenario, the code is unlikely to have hardcoded offset the way that line #323 has).

If you pass something as the `sortBy` argument of `GetPagination` method, that'd be great too, since it would demonstrate how sorting works (by passing DB column name, I suppose).